client/src/main/java/org/asynchttpclient/cookie/CookieEncoder.java

```diff
@@ -63,8 +63,7 @@ public static String encode(Collection<Cookie> cookies) {
         } else {
             Cookie[] cookiesSorted = cookies.toArray(new Cookie[cookies.size()]);
             Arrays.sort(cookiesSorted, COOKIE_COMPARATOR);
-            for (int i = 0; i < cookiesSorted.length; i++) {
-                Cookie cookie = cookiesSorted[i];
+            for (Cookie cookie : cookiesSorted) {
```

Comment:
Actually, I would tend to disagree here: the original form saves an Iterator. One could argue JIT escape analysis might be able to save it, but maybe not. Netty does the same thing.